libraries/Wire/src/utility/twi.c

```diff
@@ -856,12 +856,20 @@ i2c_status_e i2c_master_write(i2c_t *obj, uint8_t dev_address,
 #if defined(I2C_OTHER_FRAME)
     uint32_t XferOptions = obj->handle.XferOptions; // save XferOptions value, because handle can be modified by HAL, which cause issue in case of NACK from slave
 #endif
-
 #if defined(I2C_OTHER_FRAME)
-    if (HAL_I2C_Master_Seq_Transmit_IT(&(obj->handle), dev_address, data, size, XferOptions) == HAL_OK) {
+    while (HAL_I2C_Master_Seq_Transmit_IT(&(obj->handle), dev_address, data, size, XferOptions) == HAL_BUSY) {
```

Comment:
I think this work because  `HAL_I2C_Master_Seq_Transmit_IT()` and `HAL_I2C_Master_Transmit_IT()` today only returns `HAL_BUSY `or `HAL_OK`.
But no guarantee it would not return `HAL_ERROR` (like `HAL_I2C_Master_Transmit()` or H`AL_I2C_Master_Transmit_DMA()` do).

So even if it is not a problem today, it would be better to cover the case of return `HAL_ERROR`

Reply:
Done